internal/resolver/delegatingresolver/delegatingresolver_test.go

```diff
@@ -0,0 +1,306 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package delegatingresolver
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/internal/attributes"
+	"google.golang.org/grpc/internal/grpctest"
+	"google.golang.org/grpc/internal/testutils"
+	"google.golang.org/grpc/resolver"
+	_ "google.golang.org/grpc/resolver/dns" // To register dns resolver.
+	"google.golang.org/grpc/resolver/manual"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+const (
+	targetTestAddr          = "test.com"
+	resolvedTargetTestAddr  = "1.2.3.4:8080"
+	resolvedTargetTestAddr1 = "1.2.3.5:8080"
+	envProxyAddr            = "proxytest.com"
+	resolvedProxyTestAddr   = "2.3.4.5:7687"
+	resolvedProxyTestAddr1  = "2.3.4.6:7687"
+)
+
+// overwrite function overwrites HTTPSProxyFromEnvironment and
```

Comment:
This could probably be code that was moved over from the transport package. But `overwrite` is not a descriptive enough function name. Please fix.

Reply:
Done.